pontoon/static/js/tags/admin/manager.js

```diff
@@ -0,0 +1,57 @@
+
+import React from 'react';
+
+import TagResourceSearch from './search';
+import TagResourceTable from './table';
+import {ErrorList} from 'widgets/errors';
+
+import "./tag-resources.css";
+
+import {dataManager} from 'utils/data';
+
+
+export class TagResourceManagerWidget extends React.Component {
+
+    constructor (props) {
+        super(props);
+        this.state = {type: 'assoc', search: "*"};
+    }
+
+    componentDidMount () {
+        this.props.refreshData({...this.state});
+    }
+
+    componentDidUpdate (prevProps, prevState) {
+        if (prevState !== this.state) {
+            this.props.refreshData({...this.state});
+        }
+    }
+
+    handleSearchChange = (change) => {
+        return this.setState(change);
+    }
+
+    handleSubmit = (checked) => {
+        return this.props.handleSubmit(
+            Object.assign({}, {...this.state}, checked));
```

Comment:
No need to deconstruct `state` here: `Object.assign({}, this.state, checked)`.

Reply:
If object spread is available, [`Object.assign()` is not needed either](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&code_lz=FAYw9gdgzgLgBAQzgXjgbzgMzGAXHAcgCMEAnAuAXwG5RJY4iV0sd9iEAvAgGkbPYluVWnWjwQzNADpZCPrOlEaYqGAA2AU2nqwAcwAUIAJTUgA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&lineWrap=false&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0&envVersion=):
```
return {...this.state, checked}
```
That will already create and return a new object.

Reply:
ah, thanks, nice

Reply:
Oh, I never saw that syntax before! Sweet!